### PR TITLE
add Lmod hook to set  to ^smcuda when loading OpenMPI module to work around bug + renaming

### PR DESCRIPTION
**Pull request by TopRichard**
Sync NESSI software layer with EESSI software layer, EESSI PR473

### CODE REVIEW

File: create_lmodrc.py [hunk 1]
```diff
@@ -44,26 +44,26 @@
...
-    -- when loading CUDA enabled modules check if the necessary driver libraries are accessible to the EESSI linker,
+    -- when loading CUDA enabled modules check if the necessary driver libraries are accessible to the NESSI linker,
     -- otherwise, refuse to load the requested module and print error message
     local haveGpu = mt:haveProperty(simpleName,"arch","gpu")
     if haveGpu then
         local arch = os.getenv("EESSI_CPU_FAMILY") or ""
-        local cudaVersionFile = "/cvmfs/software.eessi.io/host_injections/nvidia/" .. arch .. "/latest/cuda_version.txt"
-        local cudaDriverFile = "/cvmfs/software.eessi.io/host_injections/nvidia/" .. arch .. "/latest/libcuda.so"
+        local cudaVersionFile = "/cvmfs/pilot.nessi.no/host_injections/nvidia/" .. arch .. "/latest/cuda_version.txt"
+        local cudaDriverFile = "/cvmfs/pilot.nessi.no/host_injections/nvidia/" .. arch .. "/latest/libcuda.so"
```
Comment (trz42) [inline create_lmodrc.py:60]: Looks like we overlooked this when syncing NESSI with EESSI back in January?

**Review by trz42 — COMMENTED** [review]
Content-wise I think this looks ok.

As done in https://github.com/EESSI/software-layer/pull/473, we should build it for the different architectures and ingest it (because it changes the `lmodrc.lua`). https://github.com/EESSI/software-layer/pull/473 also documents nicely how it was tested (we should not see the same message though since we don't have a `neoverse_v1` software subdirectory).

For the future, I think it would be good to make a clickable link, e.g., [EESSI/software-layer PR473](https://github.com/EESSI/software-layer/pull/473) instead of just putting in the plain number in the description.

A little suggestion would also be to apply the same commit as done in https://github.com/EESSI/software-layer/pull/473 This might be done via `git cherry-pick ...` (or some other command). Thus there is less doubt if the changes made here were identical or not to what was done in https://github.com/EESSI/software-layer/pull/473 Of course, this might not always be so easy in case of conflicts or more complex changes.

There are a number of other changes included here. They could be be done in a separate commit. Thus the commit adding the hook could be compared to the commit in https://github.com/EESSI/software-layer/pull/473.

File: create_lmodrc.py (hunk 1 above)
Comment (TopRichard) [reply create_lmodrc.py:60]: We missed this as we were not too focused on the cuda changes by then.

**Comment by trz42** [comment]
bot: build inst:Saga-NESSI repo:nessi-2023.06-swl-deb11 arch:x86_64/intel/broadwell
bot: build inst:Saga-NESSI repo:nessi-2023.06-swl-deb11 arch:x86_64/intel/skylake
bot: build inst:eX3 repo:nessi-2023.06-swl-deb11 arch:aarch64/generic
bot: build inst:eX3 repo:nessi-2023.06-swl-deb11 arch:zen2
bot: build inst:AWS-MC-NESSI repo:nessi-2023.06-swl-deb11 arch:x86_64/generic

**Comment by trz42** [comment]
Broadwell nodes on Saga are busy. Try Fram.

bot: build inst:Fram-NESSI repo:nessi-2023.06-swl-deb11 arch:x86_64/intel/broadwell

**Comment by trz42** [comment]
Builds for all architectures succeeded. Cancelling pending job on Saga.

**Comment by trz42** [comment]
Checklist before starting deployment (setting `bot:deploy` label): 

- [x]  Check if the _SPDX license identifier_ is provided
  - not needed because no software was built (just updating an Lmod config script) 
- [x]  Check whether builds for all required architectures succeed (SUCCESS message + reasonably sized tarball)
- [x]  Check if the PR is up-to-date with the target branch nessi.no-2023.06 in the repository (if not what are the differences)
- [x]  Assess if all requested changes are sound (checking files changed on GitHub.com)
- [x]  Verify that all easyconfig/s being built are included with the EB version used (if not why not)
  - no easyconfigs were used
- [x]  Review changes (if any) needed to get the build(s) succeed (common changes for all architectures, changes for a single architecture, changes because of build environment specifics, etc.)

**Comment by trz42** [comment]
Target architectures
  - `x86_64-intel-broadwell`
  - `x86_64-intel-skylake_avx512`
  - `x86_64-amd-zen2`
  - `x86_64-generic`
  - `aarch64-generic`

Checklist for deployment/ingestion
- [x] at least one tarball for each architecture has been uploaded
- [x] at least one tarball for each architecture has been staged
- [x] for each staged tarball was a PR for approval/rejection opened
- [x] for each architecture **one and only one** tarball has been approved
- [x] for each architecture **one and only one** tarball was ingested
- [x] all tests in the CI have succeeded (rerun failed tests ~ 10 mins after last ingest)
  - no rerun needed as CI is independent of changes to `lmodrc.lua`
- [x] has the package become available via CernVM-FS for all architectures ingested (for lmod cache updates pay attention to timestamp of files)
  - file `lmodrc.lua` for `cascadelake` wasn't updated (which is fine); should remove software branch for this microarchitecture
<details><summary>command & log</summary>

  **command**
  ```
  cd /cvmfs/pilot.nessi.no/versions/2023.06/software/linux ; \
  ls -l \
        x86_64/{generic,amd/zen2,intel/[bsc]*}/.lmod/lmodrc.lua \
        aarch64/generic/.lmod/lmodrc.lua
  ```
  **log**
```
-rw-r--r--. 1 cvmfs cvmfs 5645 Apr  5 20:25 aarch64/generic/.lmod/lmodrc.lua
-rw-r--r--. 1 cvmfs cvmfs 5645 Apr  5 20:25 x86_64/amd/zen2/.lmod/lmodrc.lua
-rw-r--r--. 1 cvmfs cvmfs 5643 Apr  5 20:29 x86_64/generic/.lmod/lmodrc.lua
-rw-rw-r--. 1 cvmfs cvmfs 5659 Apr  5 20:31 x86_64/intel/broadwell/.lmod/lmodrc.lua
-rw-r--r--. 1 cvmfs cvmfs  283 Jun 18  2023 x86_64/intel/cascadelake/.lmod/lmodrc.lua
-rw-r--r--. 1 cvmfs cvmfs 5669 Apr  5 20:26 x86_64/intel/skylake_avx512/.lmod/lmodrc.lua 
```
</details>

**Review by trz42 — APPROVED** [review]
All fine. Updated files ingested.

**Pull request merged**